jpos/src/main/java/org/jpos/q2/Q2.java

```diff
@@ -360,6 +360,84 @@ public static Q2 getQ2(long timeout) {
         return (Q2) NameRegistrar.get(JMX_NAME, timeout);
     }
 
+    public Map<String, Object> getXmlConfigMap() throws Exception {
+        Iterator iter = dirMap.entrySet().iterator();
+        Map<String, Object> config = new HashMap<>();
+        while (iter.hasNext()) {
+            Map.Entry entry = (Map.Entry) iter.next();
+            File   f        = (File)   entry.getKey ();
+            QEntry qentry   = (QEntry) entry.getValue ();
+            if (qentry.isQBean() && qentry.isQPersist() && isXml(f)) {
+                String fname = f.getName();
+                fname = fname.substring(0, fname.length() - 4);
+                ObjectName qname = qentry.getObjectName ();
+                Element e = (Element) server.getAttribute (qname, "Persist");
+                config.put(fname, getElementMap(e));
+            }
+        }
+        return config;
+    }
+
+    private Map<String, Object> getElementMap(Element e) {
+        Map<String, Object> map = new HashMap<>();
+        String className = e.getAttributeValue("class");
+        if (className != null) {
+            map.put("class", className);
+        }
+        List<Element> children = e.getChildren();
+        Map<String, Object> props = null;
+        Map<String, Object> attr = null;
+        for(Element child: children){
+            String key = child.getName();
+            if (key == "property") {
+                if (props == null) {
+                    props = new HashMap<>();
+                }
+                props.put(child.getAttributeValue("name"), child.getAttributeValue("value"));
+            } else if (key == "attr") {
```

Comment:
Shouldn't you use `"attr".equals(key)` here ?

Reply:
Pushed a fix changing this to` "attr".equals(key)`.